src/typescript.js

```diff
@@ -2,6 +2,11 @@
  * @type {import('eslint').Linter.Config}
  */
 export const rules = {
+  /**
+   * Conflicting rules
+   */
+  'dot-notation': 'off',
```

Comment:
I think there was a mistake in the original reported issue.

The issue mentions `@typescript-eslint/dot-notation`, but instead we always have used the normal javascript `dot-notation` rule.

The https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/dot-notation.mdx rule already supports this `allowIndexSignaturePropertyAccess` compiler flag (when enabled as an option), and recommends turning off the original rule. Seems like we never did that :)

```suggestion
  'dot-notation': 'off',
  '@typescript-eslint/dot-notation': ['error']
```

(not sure if that's the correct place, surely this is not the only rule it overrides and has to be turned off?)